app/models/container_deployment_node.rb

```diff
@@ -4,4 +4,41 @@ class ContainerDeploymentNode < ApplicationRecord
   serialize :labels, Hash
   serialize :customizations, Hash
   acts_as_miq_taggable
+
+  def node_address
+    if vm
+      vm.hostnames.last || vm.hardware.ipaddresses.last
+    elsif address
+      address
+    end
+  end
+
+  def labels_hash
+    labels ||= {}
+    labels.each do |key, val|
+      labels[key] = val
+    end
+    labels
+  end
+
+  def roles
+    roles = []
+    tags.reset
+    tags.each do |tag|
+      tag_entry = tag.name.split("/").last
+      tag_entry = "master" if tag_entry.include?("deployment_master")
+      roles << tag_entry
```

Comment:
This can be simplified with a .collect.uniq, and thus no need for the temporary roles variable.